modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+"""
+This module contains classes for easy data management
+"""
+from constants import HUMIDITY_UNIT, TEMPERATURE_UNIT, WEATHER_FILES_DIR
+from modules.utils import (
+    get_date,
+    get_highest_temperature,
+    get_lowest_temperature,
+    get_max_humidity,
+    get_mean_humidity,
+    get_month_name,
+    parse_line,
+    read_data,
+)
+
+
+class DayData:
+    """
+    This class holds data of a particular day
+    """
+
+    def __init__(self, line):
+        """
+        Initializes the line member and calls populate to initialize other members
+        Args:
+            line(list): list of strings containing all data from line read of weather file
+        """
+        self.line = line
+        self.populate()
+
+    def populate(self):
+        """
+        Sets the values of highest_temperature, lowest_temperature, max_humidity,
+        mean_humidity and date by using the parsed_line
+        Returns:
+            None
+        """
+        self.highest_temperature = get_highest_temperature(self.line)
+        self.lowest_temperature = get_lowest_temperature(self.line)
+        self.max_humidity = get_max_humidity(self.line)
+        self.mean_humidity = get_mean_humidity(self.line)
+        self.date = get_date(self.line)
+
+    def get_max_temperature(self):
```

Comment:
We don't really need these getters since we will be able to access the values directly from the instance. This goes for all the other getters in this class unless we are extending the functionality in the function e.g. we are returning the value based on some calculations.

Reply:
I have removed all the getters. and used `.` operator to get values where needed